### ISSUE TITLE
QGIS crashes on save when layout objects have been deleted.

### ISSUE THREAD

**Issue by samlalwani** [qgis/QGIS#53988]
## User Feedback

QGIS crashes on save when layout objects have been deleted.

## Report Details

**Python Stack Trace**
```
Windows fatal exception: access violation

Current thread 0x000044c8 (most recent call first):
<no Python frame>
```

**Stack Trace**

```

QgsLayoutItemAttributeTable::writePropertiesToElement :
QgsLayoutMultiFrame::writeXml :
QgsLayout::writeXml :
QgsPrintLayout::writeXml :
QgsPrintLayout::writeLayoutXml :
QgsLayoutManager::writeXml :
QgsProject::writeProjectFile :
QgsProject::zip :
QgsProject::write :
QgisApp::fileSave :
QObject::qt_static_metacall :
QAction::activate :
QAbstractButton::click :
QAbstractButton::mouseReleaseEvent :
QToolButton::mouseReleaseEvent :
QWidget::event :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QApplicationPrivate::sendMouseEvent :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify :
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processMouseEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
```

**QGIS Info**
QGIS Version: 3.32.1-Lima
QGIS code revision: 4a078ecf63
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.7.1
Running against GDAL: 3.7.1

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.19045

### Steps to reproduce the issue

Open existing layout.
delete an object
click save project
QGIS crashes 

### Versions

QGIS version
3.32.1-Lima
QGIS code revision
4a078ecf63
Qt version
5.15.3
Python version
3.9.5
GDAL/OGR version
3.7.1
PROJ version
9.2.1
EPSG Registry database version
v10.088 (2023-05-13)
GEOS version
3.12.0-CAPI-1.18.0
SQLite version
3.41.1
PDAL version
2.5.5
PostgreSQL client version
15.2
SpatiaLite version
5.0.1
QWT version
6.1.6
QScintilla2 version
2.13.1
OS version
Windows 10 Version 2009

Active Python plugins
Qgis2threejs
2.7.1
qgis_resource_sharing
1.0.0
qgis_stac
1.1.1
QuickOSM
2.2.2
quick_map_services
0.19.33
shapetools
3.4.14
db_manager
0.1.20
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by YoannQDQ** [comment]
> project file is attached

Where though? I'm sorry if I'm missing something obvious, but I don't see a link to the project anywhere in your issue.

**Comment by samlalwani** [comment]
Project file has been zipped and attached 
[SFO.zip](https://github.com/qgis/QGIS/files/12261604/SFO.zip)

**Comment by frasemcl** [comment]
I'm experiencing the same issue on 3.28.10, any idea when the fix outlined here will make it into the LTR? Thx

**Comment by YoannQDQ** [comment]
The fix was backported to 3.28, so it should be included in 3.28.11 (release date according to the [roadmap](https://www.qgis.org/fr/site/getinvolved/development/roadmap.html): 2023/09/15)
